contracts/StakingContract.sol

```diff
@@ -222,9 +258,14 @@ contract StakingContract is IStakingContract, IMigratableStakingContract {
     function withdraw() external {
         address stakeOwner = msg.sender;
 
-        (uint256 withdrawnAmount, uint256 totalStakedAmount) = withdraw(stakeOwner);
+        WithdrawResult memory res = withdraw(stakeOwner);
 
-        emit Withdrew(stakeOwner, withdrawnAmount, totalStakedAmount);
+        emit Withdrew(stakeOwner, res.withdrawnAmount, res.stakedAmount);
+
+        // Note: we aren't concerned with reentrancy since:
+        //   1. At this point, due to the CEI pattern, a reentrant notifier can't affect the effects of this method.
+        //   2. The notifier is set and managed by the migration manager.
+        stakeChange(stakeOwner, res.stakedAmountDiff, res.stakedAmountDiffSign, res.stakedAmount);
```

Comment:
stakeChange notify should not be triggered by the external withdraw that does not change the stake